File: src/Module/TicketAttribute/TicketAttributeEntity.java
package Module.TicketAttribute;

import Module.Ticket.TicketEntity;
import Module.Ticket.TicketModel;

import java.io.Serializable;

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
public class TicketAttributeEntity implements Serializable {
    public int id;
    public String status;
    public String priority;
    public String rating;
    public String reopened;
    public Integer ticketId;
    public TicketEntity ticketEntity;


    public TicketAttributeEntity() {
    }

    public TicketAttributeEntity(TicketattributeModel TicketattributeModel, Object... objects) {
        this.id = TicketattributeModel.getId();
        this.status = TicketattributeModel.getStatus();
        this.priority = TicketattributeModel.getPriority();
        this.rating = TicketattributeModel.getRating();
        this.reopened = TicketattributeModel.getReopened();
        this.ticketId = TicketattributeModel.getTicketId();
        for (Object object : objects) {
            if (object instanceof TicketModel) {
                this.ticketEntity = new TicketEntity((TicketModel) object);
            }
        }
    }

    public TicketattributeModel toModel() {
        TicketattributeModel ticketattributeModel = new TicketattributeModel();
        ticketattributeModel.setId(id);
        ticketattributeModel.setStatus(status);
        ticketattributeModel.setPriority(priority);
        ticketattributeModel.setRating(rating);
        ticketattributeModel.setReopened(reopened);
        ticketattributeModel.setTicketId(ticketId);
        return ticketattributeModel;
    }
}

File: web/Angular4/src/app/Modules/TicketAttribute/TicketAttribute.Entity.ts
import {TicketEntity} from "../Ticket/Ticket.Entity";

export class TicketAttributeEntity {

    id: number;
    status: string;
    priority: string;
    rating: string;
    reopened : string;
    ticketId : number;
    ticketEntity : TicketEntity;


    IsEdit: boolean;
    IsActive: boolean = false;
    IsSelected: boolean = false;
    
    constructor(data: any = null) {
        if (data == null) {
            this.id = null;
            this.status = null;
            this.priority = null;
            this.rating = null;
            this.reopened = null;
            this.ticketId = null;
            this.ticketEntity = new TicketEntity();
        } else {
            this.id = data.id;
            this.status = data.status;
            this.priority = data.priority;
            this.rating = data.rating;
            this.reopened = data.reopened;
            this.ticketId = data.ticketId;
            this.ticketEntity = new TicketEntity();

            if (this.ticketEntity == null) this.ticketEntity = new TicketEntity();
        }
        this.IsEdit = false;
    }
}